### PR TITLE
WDPD-207: Update to wpp v2

### PR DESCRIPTION
**Pull request by ps-amaureder**
### This PR

* Changes from wpp v1 to wpp v2

### Notes

* Base branch: wdpd-207-base 

### How to Test

* Test payment Credit Card payment method

### Jira Links
https://jira.parkside.at/browse/WDPD-207

### CODE REVIEW

File: Model/CreditCardPaymentMethod.php [hunk 1]
```diff
@@ -148,6 +148,23 @@ private function _convertAmountCurrency($fAmount, $fFromFactor, $oToCurrency)
      */
     public function getConfigFields()
     {
+        $iUrlFieldsOffset = 1;
```
Comment (ps-holler) [inline Model/CreditCardPaymentMethod.php]: I think this should be in a helper, e.g.:

```php
public static function insertToArrayAtPosition(&$aArray, $aContents, $iIndex)
{
    $aArray = array_merge(
        array_slice($aArray, 0, $iIndex, true),
        $aContents,
        array_slice($aArray, $iIndex, count($aArray) - 1, true)
    );
}
```

File: Extend/Controller/Admin/PaymentMain.php [hunk 2]
```diff
@@ -247,15 +280,16 @@ private function _validateRequestParameters($aParams)
...
+        if ($this->_isCredentialsSet($sUrl, $sUser, $sPass)) {
             $oConfig = new Config($sUrl, $sUser, $sPass);
             $oTransactionService = $this->_getTransactionService($oConfig);
             return $oTransactionService->checkCredentials();
         }
 
         // handle the case of different configuration per currency (currently only Payolution)
         // there it's only checked if config values are entered on the frontend but no validation is done on the backend
-        return $this->_checkCurrencyConfigFields($aParams);
+        return $aParams['oxpayments__oxid'] === PayolutionInvoicePaymentMethod::getName(true)
```
Comment (ps-cgrach) [inline Extend/Controller/Admin/PaymentMain.php]: please, move the payment check to the method, and do an early `return true;`

File: Extend/Controller/Admin/PaymentMain.php [hunk 3]
```diff
@@ -370,4 +404,76 @@ private function _getTransactionService($oConfig)
...
+     *
+     * @param array $aParams
+     *
+     * @return bool
+     *
+     * @since 1.3.0
+     */
+    private function _noCreditCardUrlIsTest($aParams)
+    {
+        return !($this->_isTestApiUrl($aParams) || $this->_isTestApiUrlWpp($aParams));
```
Comment (ps-cgrach) [inline Extend/Controller/Admin/PaymentMain.php]: if you pass the URL right away, you can get rid of one method.

File: Model/CreditCardPaymentMethod.php (hunk 1 above)
Comment (ps-cgrach) [reply Model/CreditCardPaymentMethod.php]: :+1:

File: Extend/Controller/Admin/PaymentMain.php (hunk 2 above)
Comment (ps-amaureder) [reply Extend/Controller/Admin/PaymentMain.php]: simplified the code

File: Extend/Controller/Admin/PaymentMain.php [hunk 4]
```diff
@@ -263,15 +296,19 @@ private function _validateRequestParameters($aParams)
         $sUser = $aParams['oxpayments__wdoxidee_httpuser'];
         $sPass = $aParams['oxpayments__wdoxidee_httppass'];
 
-        if ($sUrl && $sUser && $sPass) {
+        if ($aParams['oxpayments__oxid'] === PayolutionInvoicePaymentMethod::getName(true)) {
```
Comment (ps-dfritzsch) [inline Extend/Controller/Admin/PaymentMain.php]: @tomazpu this Payolution check has been existing before (see few lines below), but had to be moved in its own if-statement to make WPP v2 work as expected.

**Pull request merged**